src/main/java/org/ihtsdo/buildcloud/core/service/InputFileServiceImpl.java

```diff
@@ -325,12 +325,13 @@ public void copyExternallyMaintainedFiles(String releaseCenterKey, String source
 					InputStream inputStream = fileHelper.getFileStream(sourceFilePath);
 					try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
 						 PrintWriter writer = new PrintWriter(new BufferedOutputStream(new FileOutputStream(tmpFile)))) {
-						String str;
-						boolean copyHeaderComplete = false;
-						while ((str = reader.readLine()) != null) {
-							if (str != null && !copyHeaderComplete) {
-								writer.println(str);
-								copyHeaderComplete = true; // should not we break out of while here?
+						String str = reader.readLine();
+						if (str != null) {
+							writer.println(str);
+							if (!isHeaderOnly) {
+								while ((str = reader.readLine()) != null) {
+									writer.println(str);
+								}
```

Comment:
@astro-snail  Yes the logic looks good now. This method is only used when start a new authoring cycle and it is not part of input files processing. It is better to restore the original class. Also it is a bad idea to just log an exception.

Reply:
@CoderMChu Do you mean to restore `ExternallyMaintainedRefSetsServiceImpl` class? Or move this method to `ReleaseServiceImpl` class where it originally was?
Without ExternallyMaintainedRefSetsService I needed a file helper to copy files so I moved it out of `ReleaseServiceImpl` because it has no file helper. Shall I get a constructor and file helper in `ReleaseServiceImpl`  or rollback everything and restore `ExternallyMaintainedRefSetsServiceImpl` and it's interface? Let's have a quick call. Thank you!